src/utils/client.js

```diff
@@ -273,14 +283,22 @@ export default class Client {
     });
   }
 
-  submitCustom(customKeyValues) {
+  async submitCustom(customKeyValues) {
     const { userKey } = initInfo();
     const url = `${API_URL}panel/custom/submit/v2/`;
 
     const jsonKeyVals = {};
     const formData = new FormData();
-    Object.entries(customKeyValues).forEach(([key, val]) => {
+    const promiseResults = await Promise.all(
+      Object.entries(customKeyValues).map(([key, val]) =>
+        Promise.all([key, Promise.resolve(val)])
```

Comment:
whoops, i didn't realize you'd need to wrap promises like this. i no longer fele strongly about using this approach or the previous